src/pd/async/client.rs

```diff
@@ -322,19 +322,28 @@ impl PdClient for RpcClient {
         Ok(resp.take_region())
     }
 
-    fn get_region_by_id(&self, region_id: u64) -> Result<Option<metapb::Region>> {
+    fn get_region_by_id(&self, region_id: u64) -> PdFuture<Option<metapb::Region>> {
         let mut req = pdpb::GetRegionByIDRequest::new();
         req.set_header(self.header());
         req.set_region_id(region_id);
 
-        let mut resp = try!(do_request(self, |client| client.GetRegionByID(req.clone())));
-        try!(check_resp_header(resp.get_header()));
+        let request_factory = |client: &PDAsyncClient, req: pdpb::GetRegionByIDRequest| {
```

Comment:
What it returns is future rather than request. The request is req defined at L326.